server/repository/src/db_diesel/temperature_breach_config_row.rs

```diff
@@ -0,0 +1,114 @@
+use super::{
+    store_row::store,
+    temperature_breach_config_row::temperature_breach_config::dsl as temperature_breach_config_dsl,
+    temperature_breach_row::TemperatureBreachRowType, StorageConnection,
+};
+
+use crate::repository_error::RepositoryError;
+use diesel::prelude::*;
+
+table! {
+    temperature_breach_config (id) {
+        id -> Text,
+        duration -> Integer,
+        #[sql_name = "type"] type_ -> crate::db_diesel::temperature_breach_row::TemperatureBreachRowTypeMapping,
+        description -> Text,
+        is_active -> Bool,
+        store_id -> Nullable<Text>,
+        minimum_temperature -> Double,
+        maximum_temperature -> Double,
+    }
+}
+
+table! {
+    #[sql_name = "temperature_breach_config"]
+    temperature_breach_config_is_sync_update (id) {
+        id -> Text,
+        is_sync_update -> Bool,
+    }
+}
+
+joinable!(temperature_breach_config -> store (store_id));
+
+#[derive(Clone, Queryable, Insertable, AsChangeset, Debug, PartialEq)]
```

Comment:
Just with the `Defaults`, we started using them due to test cases, had a lot of test data that needed to be updated if new field was added, but if we use default for all fields apart from the ones being tested, it makes it much easier. Anyways the auto derive `Default` works if all of struct fields have default implementation, in this case the enum doesn't but we recently discovered `#[default]`` annotation which makes it slightly easier (btw there are other types where we have to manually implement Default, because field type doesn't implement Default and it's foreign crate type, for which we can't manually implement Default, like NaiveDate), anyways, could do this:

Reply:
https://github.com/openmsupply/open-msupply/compare/1955-temperature-sensor-schema-for-temperature-logs-and-breaches...1955-temperature-sensor-schema-for-temperature-logs-and-breaches(Derive-default-example)

Reply:
Nice! I've done that for all of them: https://github.com/openmsupply/open-msupply/pull/2132/commits/8bc9323c5d6c4a06851d3b1bd84a47fb69bbe489